tests/e2e/test_load.py

```diff
@@ -0,0 +1,49 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+import torch
+
+from azstoragetorch.io import BlobIO
+
+
+@pytest.fixture(scope="module", autouse=True)
+def torch_hub_cache(tmp_path_factory):
+    current_dir = torch.hub.get_dir()
+    yield tmp_path_factory.mktemp("torch_hub")
+    torch.hub.set_dir(current_dir)
+
+
+@pytest.fixture(scope="module")
+def model(torch_hub_cache):
+    torch.hub.set_dir(torch_hub_cache)
+    model = torch.hub.load("pytorch/vision:v0.10.0", "resnet101", pretrained=False)
+    return model
+
+
+@pytest.fixture(scope="module")
+def upload_model(model, container_client, torch_hub_cache):
+    blob_name = torch_hub_cache / "model.pth"
```

Comment:
Instead of using `torch_hub_cache`, let's use `tmp_path` directly instead. My main concern in using the cache directly to save models is that `torch.hub` may be expecting a specific type of directory structure for its local cache and we may cause issues at some point in directly writing to it. By writing to a file outside of the torch hub cache, we eliminate that possibility.